cmd/src/extensions_publish.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+)
+
+func init() {
+	usage := `
+Publish an extension to Sourcegraph, creating it (if necessary).
+
+Examples:
+
+  Publish the "alice/myextension" extension described by package.json in the current directory:
+
+    	$ cat package.json
+        {
+          "name":      "myextension",
+          "publisher": "alice",
+          "title":     "My Extension",
+          "url":       "https://example.com/bundled-extension.js"
+        }
+    	$ src extensions publish
+
+`
+
+	flagSet := flag.NewFlagSet("publish", flag.ExitOnError)
+	usageFunc := func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src extensions %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+	var (
+		extensionIDFlag = flagSet.String("extension-id", "", `Override the extension ID in the manifest. (default: read from -manifest file)`)
+		manifestFlag    = flagSet.String("manifest", "package.json", `The extension manifest file.`)
+		forceFlag       = flagSet.Bool("force", false, `Force publish the extension, even if there are validation problems or other warnings.`)
+		apiFlags        = newAPIFlags(flagSet)
+	)
+
+	handler := func(args []string) error {
+		flagSet.Parse(args)
+
+		manifest, err := ioutil.ReadFile(*manifestFlag)
+		if err != nil {
+			return fmt.Errorf("%s\n\nRun this command in a directory with a %s file for an extension.\n\nSee 'src extensions %s -h' for help.", err, *manifestFlag, flagSet.Name())
```

Comment:
Return a `usageError` here instead, and it'll print the help output which is useful here: https://github.com/sourcegraph/src-cli/blob/056c528f8828f7cfbedb4e1b7a7572bb31ccea7f/cmd/src/search.go#L60

Also, what if they just specified the wrong `-manifest` path file? It seems like odd wording in this case. Maybe check if `-manifest` is the default and change wording based on that.

Reply:
Ah, got it, okay. LGTM then